src/components/duelling-picklist/picklist/picklist.component.js

```diff
@@ -1,30 +1,38 @@
-import React, { useMemo, useCallback } from "react";
+import React, { useCallback, useContext, useEffect, useMemo } from "react";
 import PropTypes from "prop-types";
 import { TransitionGroup } from "react-transition-group";
 
-import Events from "../../../utils/helpers/events";
 import { StyledPicklist, StyledEmptyContainer } from "./picklist.style";
+import FocusContext from "../duelling-picklist.context";
+import Events from "../../../utils/helpers/events";
+import PicklistGroup from "../picklist-group/picklist-group.component";
 
-export const Picklist = ({ disabled, children, placeholder }) => {
-  const isEmpty = useMemo(() => !React.Children.toArray(children).length > 0, [
+export const Picklist = ({ disabled, children, placeholder, index }) => {
+  const { elementToFocus, setElementToFocus } = useContext(FocusContext);
+
+  const isEmpty = useMemo(() => !React.Children.toArray(children).length, [
     children,
   ]);
 
+  const filteredChildren = React.Children.toArray(children).filter(
+    (child) => child
+  );
+
   const refs = useMemo(
     () =>
       Array.from(
         {
-          length: React.Children.count(children),
+          length: filteredChildren.length,
         },
         () => React.createRef()
       ),
-    [children]
+    [filteredChildren.length]
   );
 
```

Comment:
both functions are the same as above, maybe it will be good to create a `picklist.utils.js` file and move it there? Just an idea

Reply:
I originally considered it but as it's only this one function I decided it was overkill and I wouldn't be able to `useMemo` which I think in this case is worth it